### ISSUE TITLE
Enhance keyboard functionality for multi-select

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#1277]
## Summary 
Add the optional keyboard combinations recommended by W3C for multi-select components to make it easier for a user to select various options via the keyboard.

## 💬 Description 
W3C provide a [list of keyboard combinations](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/#:~:text=of%20characters%20typed.-,Multiple%20Selection,-%3A%20Authors%20may%20implement) which they recommend should be included when creating a multi-select component. This ticket is only for the optional ones; all of the essential keyboard behaviour has been done as part of the multi-select ticket (#257), as well as the Ctrl+A / Cmd+A functionality to select or deselect all options.

## 💰 User value 
This will increase the number of actions a user can perform when interacting with the multi-select using a keyboard.

## 📝 Acceptance Criteria  
The following acceptance criteria have been separated out of the multi-select ticket.

*Given* that focus is applied to the droplist attached to a multi-select input
*When* I use the 'up/down' keys, and hold 'shift'
*Then* the options should be focussed in order in the direction of the arrow keys,
*And* the option should automatically be selected/deselected. 

*Given* that focus is applied to the droplist attached to a mutli-select input
*When* an option is selected (by mouse or keyboard) while holding the 'shift' key
*Then* all contiguous options between the current option and the last selected option should be selected. 

*Given* that focus is applied to an option in an expanded droplist
*When* using the key combination 'ctrl+shift+home'
*Then* all options from the current option to the first option in the list should be selected. 

*Given* that focus is applied to an option in an expanded droplist
*When* using the key combination 'ctrl+shift+end'
*Then* all options from the current option to the last option in the list should be selected. 

## Additional info 
Blocked by #257.

**Comment by GCHQ-Developer-847** [comment]
All functionality mentioned in the acceptance criteria in the ticket are now completed (the branch is [feature/2-1277-enhance-multi-select-keyboard-functionality](https://github.com/mi6/ic-ui-kit/compare/develop...feature/2-1277-enhance-multi-select-keyboard-functionality)). I have tried to make everything match the [W3C multi-select behaviour](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-rearrangeable/#:~:text=Last%20change%3A-Example%202%3A%20Multi%2DSelect%20Listbox-Open%20In%20CodePen) as much as possible.

I have checked that my changes don’t affect / break any behaviour on the single and searchable selects. I did this by both removing line 94 in `story-data.ts`, and changing it to “searchable”, and then playing around with different keyboard behaviour etc. All existing E2E tests for `ic-select-with-multi` pass as well. 

<br />

I have been starting to look at adding tests.

It is not possible to run unit tests in the `canary-web-components` folder so I have had to temporarily move the unit tests for ic-select-with-multi to the `web-components` folder, where they can be run (this change is in [feature/3-1277-enhance-multi-select-keyboard-functionality](https://github.com/mi6/ic-ui-kit/compare/develop...feature/3-1277-enhance-multi-select-keyboard-functionality)).

The next steps are to write the unit tests to achieve good coverage of my changes. Once they all look good, move these new tests over to where they should be in `ic-menu-with-multi/test/basic/ic-menu.spec.tsx` in the `canary-web-components` folder.

<br />

E2E tests will also need adding.

The E2E tests for this ticket may as well be done using Cypress (the existing tests for the multi-select functionality have not been done using Cypress but this will be completed as part of https://github.com/mi6/ic-ui-kit/issues/1700 - so creating these new Cypress tests will involve creating the new test files in `canary-react`).

<br />

My commits will need squashing / tidying up as well.

**Comment by GCHQ-Developer-847** [comment]
Just to say that my above comment may be a bit confusing, especially as some others may not have had as much experience as me with working with canary components.

This ticket is a medium priority so if we decide it's best we wait until I am available to pick it back up again, instead of someone else carrying on with it, I would be fine with that :)

**Comment by GCHQ-Developer-299** [comment]
> Just to say that my above comment may be a bit confusing, especially as some others may not have had as much experience as me with working with canary components or the multi-select component.
> 
> This ticket is a medium priority so if we decide it's best we wait until I am available to pick it back up again, instead of someone else carrying on with it, I would be fine with that :)

I did take a bit of a look at the branch with unit tests on it (feature/3...) and couldn't work out what additional tests needed to be written, I think i agree that it'd be useful to wait until you're back (unless a more clued in developer comes across this)
